lib/gui_classes.py

```diff
@@ -0,0 +1,261 @@
+        self.message.pack(padx=10, pady=10)
+
+
+class BpmPopup(tk.Toplevel):
+    """A GUI class representing a dialog window. The dialog asks for BPM value and checks its validity. If valid, sets BPM.
+    """
+
+    def __init__(self, master: AbstractLemApp, **kwargs: Any) -> None:
+        """Build a new BpmPopup.
+
+        Args:
+            master (AbstractLemApp): The parent widget. Must be an instance of Lem class (the top level gui class of this project).
+        """
+        super().__init__(master, **kwargs)
+        self.master: AbstractLemApp = master
+
+        self.title("enter BPM")
+
+        self._entry_line = tk.Frame(master=self)
+        self._message = tk.Label(
+            master=self._entry_line, text="Insert the desired BPM value: ")
+        self._bpm_entry = tk.Entry(master=self._entry_line, width=3)
+
+        self._message.pack(side="left")
+        self._bpm_entry.pack(side="right")
+        self._entry_line.pack(side="top", padx=30, pady=5)
+
+        self._instructions = tk.Label(
+            master=self, text="BPM value must be a whole number between 1 and 400")
+        self._instructions.pack()
+
+        self._confirm = tk.Button(
+            master=self, text="Confirm!", command=self.set_bpm)
+        self._confirm.pack(side="bottom", padx=5, pady=5)
+        self.bind(sequence="<Return>", func=lambda event: self.set_bpm())
+
+    def validate(self, value: str) -> bool:
+        """A method evaluating whether the value meets the conditions for suitable BPM (an integer between 1 and 400)
+
+        Args:
+            value (str): value taken from an entry
+
+        Returns:
+            bool: True if value meets the conditions, else False.
+        """
+        try:
+            val = int(value)
+        except ValueError:
+            return False
+
+        return val > 0 and val <= 400
+
+    def set_bpm(self) -> None:
+        """Try whether the entry value is a suitable BPM value. If yes, set it as a BPM.
+        """
+        entry_value = self._bpm_entry.get()
+        if not self.validate(value=entry_value):
+            return
+        bpm = int(entry_value)
+        self.master.set_bpm(bpm=bpm)
+        self.destroy()
+
+
+class AppBar(tk.Frame):
+    """A GUI class representing an app bar (the topmost widget on the page). 
+    Contains label displaying the set BPM, and initially also a button enabling to set it. 
+    This button is destroyed afterwards.
+    """
+
+    def __init__(self, master: AbstractLemApp, **kwargs: Any) -> None:
+        """Build a new AppBar.
+
+        Args:
+            master (AbstractLemApp): The parent widget. Must be an instance of Lem class (the top level gui class of this project).
+        """
+        super().__init__(master, **kwargs)
+        self.master: AbstractLemApp = master
+
+        self._bpm_lbl = tk.Label(master=self, text="BPM: ")
+        self._bpm_lbl.pack(side="left", padx=5, pady=5)
+
+        self._dialog_button = tk.Button(
```

Comment:
tohle je zvlastni formatovani zdrojaku: zacit zavorku, odsadit na dalsi radek a na nem zavorku ukoncit, bud vsechno na jeden radek, a nebo kdyz tady zacinat zavorku, tak tu zaviraji taky na samostatnem radku (za obsahem)

Reply:
Tohle formátování u mne defaultně dělá jedno rozšíření, autopep8. Je to tedy nějak normované, jednotné v rámci projektu... a nenašel jsem u rozšíření nějakou možnost tohle snadno změnit. Rád bych to nechal, přijde vám to použitelné?

Reply:
pro me tedy nezvyk, ale nechme to tak, je to detail